packages/canon/src/components/icon/icon.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { useIcons, IconNames } from './context';
+
+/** @public */
+export const Icon = ({ name }: { name: IconNames }) => {
+  const { icons } = useIcons();
+
+  const LucideIcon = icons[name];
+
+  if (!LucideIcon) {
+    console.error(`Icon "${name}" not found.`);
+    return <svg />; // Return default icon perhaps?
```

Comment:
Yep might be useful to have a little placeholder icon to communicate that the desired one wasn't found, and avoiding that being something that you'd ever actually want to use if possible 😅